Marksman/Server.fs

```diff
@@ -762,11 +763,30 @@ type MarksmanServer(client: MarksmanClient) =
         let codeActions =
             match toc with
             | None -> Array.empty
-            | Some (render, existing) ->
-                match existing with
-                | None -> [| codeAction "Create a Table of Contents" (editAt None render) |]
-                | Some (oldTocRange) ->
-                    [| codeAction "Update the Table of Contents" (editAt (Some oldTocRange) render) |]
+            | Some (render, insertion, existing) ->
+                let name =
+                    match existing with
+                    | None -> "Create a Table of Contents"
+                    | _ -> "Update the Table of Contents"
+
+                let insertionPoint =
+                    match existing with
+                    | Some (range) -> Replacing range
+                    | None -> insertion
+
+                let text =
+                    match insertionPoint with
+                    | Replacing _ -> render
+                    | After _ -> NewLine + render + NewLine
```

Comment:
If inserting after a certain range, we need to make sure to add newlines around it